applications/tari_console_wallet/src/automation/commands.rs

```diff
@@ -471,8 +477,32 @@ pub async fn command_runner(
                 let utxos = output_service.get_unspent_outputs().await?;
                 let count = utxos.len();
                 let sum: MicroTari = utxos.iter().map(|utxo| utxo.value).sum();
-                for (i, utxo) in utxos.iter().enumerate() {
-                    println!("{}. Value: {} {}", i + 1, utxo.value, utxo.features);
+                if parsed.args.is_empty() {
+                    for (i, utxo) in utxos.iter().enumerate() {
+                        println!("{}. Value: {} {}", i + 1, utxo.value, utxo.features);
+                    }
+                } else if let ParsedArgument::CSVFile(file) = parsed.args[0].clone() {
+                    let factory = PedersenCommitmentFactory::default();
+                    let file = File::create(file).map_err(|e| CommandError::File(e.to_string()))?;
+                    let mut csv_file = LineWriter::new(file);
+                    csv_file
+                        .write_all(b"\"#\",\"Value (uT)\",\"Spending Key\",\"Commitment\",\"Features\"\n")
+                        .map_err(|e| CommandError::File(e.to_string()))?;
+                    for (i, utxo) in utxos.iter().enumerate() {
+                        let line = format!(
+                            "\"{}\",\"{}\",\"{}\",\"{}\",\"{}\"\n",
+                            i + 1,
+                            utxo.value.0,
+                            utxo.spending_key.to_hex(),
+                            utxo.as_transaction_input(&factory, OutputFeatures::default())
+                                .commitment
+                                .to_hex(),
+                            utxo.features
+                        );
+                        csv_file
+                            .write_all(line.as_bytes())
+                            .map_err(|e| CommandError::File(e.to_string()))?;
```

Comment:
Same as above - using writeln! is better because it means that you do not need to allocate an extra String on the heap (obviously, in this part of the code it doesn't matter at all but thought I'd mention it)
```suggestion
                        writeln!(
                            csv_file,
                            r##""{}","{}","{}","{}","{}""##,
                            i + 1,
                            utxo.value.0,
                            utxo.spending_key.to_hex(),
                            utxo.as_transaction_input(&factory, OutputFeatures::default())
                                .commitment
                                .to_hex(),
                            utxo.features
                        )
                            .map_err(|e| CommandError::File(e.to_string()))?;
```